test/Sdk.Generator.Tests/FunctionExecutor/FunctionExecutorGeneratorTests.cs

```diff
@@ -502,6 +502,77 @@ public async ValueTask ExecuteAsync(FunctionContext context)
                 buildPropertiesDictionary: buildPropertiesDict);
         }
 
+        [Fact]
+        public async Task ClassWithSameNameAsNamespace()
+        {
+            const string inputSourceCode = @"
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Azure.Storage.Queues.Models;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+namespace TestProject
+{
+    public class TestProject
+    {
+        [Function(""FunctionA"")]
+        public HttpResponseData Foo([HttpTrigger(AuthorizationLevel.User, ""get"")] HttpRequestData r, FunctionContext c)
+        {
+            return r.CreateResponse(System.Net.HttpStatusCode.OK);
+        }
+    }
```

Comment:
Thank you! Can we also throw in a static class and static function to make sure that it also works?
```csharp
public static class Foo
{
    [Function("Foo")]
    public static HttpResponseData Run([HttpTrigger(AuthorizationLevel.Function, "get")] HttpRequestData req)
    {
        throw new NotImplementedException();
    }
}

````